File: code/javascript-for-csharp-java/compare/L02Numbers.java
// Lesson 2, the Java side: java L02Numbers.java (Java 25)
import java.math.BigInteger;

void main() {
    IO.println("0.1 + 0.2              " + (0.1 + 0.2));
    IO.println("7 / 2                  " + (7 / 2));
    IO.println("7.0 / 2                " + (7.0 / 2));
    IO.println("1.0 / 0                " + (1.0 / 0));
    IO.println("NaN == NaN             " + (Double.NaN == Double.NaN));
    int zero = 0;
    try {
        IO.println(1 / zero);
    } catch (ArithmeticException e) {
        IO.println("1 / zero               " + e);
    }
    int max = Integer.MAX_VALUE;
    IO.println("Integer.MAX_VALUE + 1  " + (max + 1));
    IO.println("Long.MAX_VALUE         " + Long.MAX_VALUE);
    IO.println("BigInteger 2^64        " + BigInteger.TWO.pow(64));
    IO.println("\"1\" + 2                " + ("1" + 2));
    IO.println("guitar emoji .length() " + "🎸".length());
    IO.println("-0.0 == 0.0            " + (-0.0 == 0.0));
}

File: code/javascript-for-csharp-java/compare/l02_numbers.cs
// Lesson 2, the C# side: dotnet run l02_numbers.cs (.NET 10)
using System.Globalization;
using System.Numerics;

CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;

Console.WriteLine($"0.1 + 0.2              {0.1 + 0.2}");
Console.WriteLine($"7 / 2                  {7 / 2}");
Console.WriteLine($"7.0 / 2                {7.0 / 2}");
Console.WriteLine($"1.0 / 0                {1.0 / 0}");
double nan = 0.0 / 0.0;
Console.WriteLine($"NaN == NaN             {nan == 0.0 / 0.0}");
int zero = 0;
try
{
    Console.WriteLine(1 / zero);
}
catch (DivideByZeroException e)
{
    Console.WriteLine($"1 / zero               {e.GetType().Name}: {e.Message}");
}
int max = int.MaxValue;
Console.WriteLine($"int.MaxValue + 1       {unchecked(max + 1)}");
Console.WriteLine($"long.MaxValue          {long.MaxValue}");
Console.WriteLine($"BigInteger.Pow(2, 64)  {BigInteger.Pow(2, 64)}");
Console.WriteLine($"\"1\" + 2                {"1" + 2}");
Console.WriteLine($"guitar emoji .Length    {"🎸".Length}");
Console.WriteLine($"-0.0 == 0.0            {-0.0 == 0.0}");
